tests/test_stream_daq.py

```diff
@@ -114,6 +116,29 @@ def test_csv_output(tmp_path, default_streamdaq, write_metadata, caplog):
         assert not output_csv.exists()
 
 
+@contextmanager
+def suppress_stdout(tmp_path):
+    """
+    Context manager to suppress stdout during a test
+    """
+    temp_file_path = tmp_path / "fake_stdout.txt"
+    with open(temp_file_path, 'w') as tempf:
+        old_stdout = sys.stdout
+        sys.stdout = tempf
+        try:
+            yield
+        finally:
+            sys.stdout = old_stdout
+
+@pytest.mark.timeout(5)
+def test_continuous_run(tmp_path, default_streamdaq):
+    """
+    Make sure continuous mode runs forever (so ends up as a timeout in the test)
+    """
+    with pytest.raises(TimeoutError):
+        with suppress_stdout(tmp_path):
+            default_streamdaq.capture(source="fpga", show_video=False, continuous=True)
+
```

Comment:
The test runs until the console is filled with log data. Replacing `sys.stdout` in the test didn't fix this, and I guess it's because it doesn't fix `stdout` in the other processes.
Things to try later:
- Don't use pytest timeout. Make a process check and termination method to `StreamDaq`, check after a while, and pass test if the processes are still alive. 
- Locally change log level params with `contextmanager`.

Reply:
I'm doing the second, and it kind of works, but it also revealed that the termination logic isn't working correctly in the test. We can force quit in practice, or terminate every process directly by catching exceptions in each method but would be better if this works correctly.